omx/gstomxaacdec.c

```diff
@@ -79,87 +78,125 @@ gst_omx_aac_dec_class_init (GstOMXAACDecClass * klass)
 static void
 gst_omx_aac_dec_init (GstOMXAACDec * self)
 {
-  /* FIXME: Other values exist too! */
-  self->spf = 1024;
+  g_return_val_if_fail (self, NULL);
+  self->spf = GST_OMX_AAC_DEC_OUTBUF_NSAMPLES;
 }
 
 static gboolean
 gst_omx_aac_dec_set_format (GstOMXAudioDec * dec, GstOMXPort * port,
     GstCaps * caps)
 {
-  GstOMXAACDec *self = GST_OMX_AAC_DEC (dec);
+  GstOMXAACDec *self = NULL;
   OMX_PARAM_PORTDEFINITIONTYPE port_def;
   OMX_AUDIO_PARAM_AACPROFILETYPE aac_param;
   OMX_ERRORTYPE err;
   GstStructure *s;
   gint rate, channels, mpegversion;
   const gchar *stream_format;
 
+  g_return_val_if_fail (dec, NULL);
```

Comment:
This should give you a warning, does this compile? This function returns `gboolean`, you are trying to return `NULL`

Reply:
Ready for review: 0947ed4a695b58965705a115cfebaebbfdd2f0db